api/app/wildfire_one/schema_parsers.py

```diff
@@ -173,6 +174,168 @@ def calculate_intensity_group(hfi: float) -> int:
     return 5
 
 
+class FireBehaviourPrediction:
+    """ Structure for storing fire behaviour prediction data. """
+
+    def __init__(self, ros: float,
+                 hfi: float, intensity_group,
+                 sixty_minute_fire_size: float,
+                 fire_type) -> None:
+        self.ros = ros
+        self.hfi = hfi
+        self.intensity_group = intensity_group
+        self.sixty_minute_fire_size = sixty_minute_fire_size
+        self.fire_type = fire_type
+
+
+def calculate_fire_behaviour_prediction_using_cffdrs(  # pylint: disable=too-many-arguments
+        latitude: float,
+        longitude: float,
+        elevation: float,
+        fuel_type: FuelTypeEnum,
+        bui: float,
+        ffmc: float,
+        cc: float,  # pylint: disable=invalid-name
+        pc: float,  # pylint: disable=invalid-name
+        wind_speed: float,
+        isi: float,
+        pdf: float,
+        cbh: float,
+        cfl: float):
+    """ Calculates fire behaviour prediction using CFFDRS. """
+    # pylint: disable=too-many-locals
+
+    # set default values in case the calculation fails (likely due to missing data)
+    fmc = cffdrs.foliar_moisture_content(latitude, longitude, elevation, get_julian_date_now())
+    sfc = cffdrs.surface_fuel_consumption(fuel_type, bui, ffmc, pc)
+    if cc is None:
+        cc = FUEL_TYPE_DEFAULTS[fuel_type]["CC"]
+
+    ros = cffdrs.rate_of_spread(FuelTypeEnum[fuel_type], isi, bui, fmc, sfc, pc=pc,
+                                cc=cc,
+                                pdf=pdf,
+                                cbh=cbh)
+    if sfc is not None:
+        cfb = calculate_cfb(FuelTypeEnum[fuel_type], fmc, sfc, ros, cbh)
+
+    if ros is not None and cfb is not None and cfl is not None:
+        hfi = cffdrs.head_fire_intensity(fuel_type=FuelTypeEnum[fuel_type],
+                                         percentage_conifer=pc,
+                                         percentage_dead_balsam_fir=pdf,
+                                         ros=ros, cfb=cfb, cfl=cfl, sfc=sfc)
+
```

Comment:
There is an intensity calculation in the spreadsheet that uses 300 as a constant; just wanted to double check that we are not using that at all?

Reply:
The intensity calculation in the spreadsheet is identical to the cffdrs calculation in FIcalc `FI   <- 300 * FC * ROS` so that part is deferred to the cffdrs package.